### ISSUE TITLE
test_get_font (idlelib.idle_test.test_config.IdleConfTest) failure on x86 Windows7 3.x

### ISSUE THREAD

**Issue by vstinner** [python/cpython#75151]
BPO | [30968](https://bugs.python.org/issue30968)
--- | :---
Nosy | @terryjreedy, @pfmoore, @vstinner, @tjguk, @zware, @zooba, @mlouielu
PRs | <li>python/cpython#2753</li><li>python/cpython#2769</li><li>python/cpython#5508</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/terryjreedy'
closed_at = <Date 2017-07-20.05:02:15.289>
created_at = <Date 2017-07-19.11:28:53.389>
labels = ['expert-IDLE', '3.7', 'tests', 'OS-windows']
title = 'test_get_font (idlelib.idle_test.test_config.IdleConfTest) failure on x86 Windows7 3.x'
updated_at = <Date 2018-02-03.05:52:36.393>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2018-02-03.05:52:36.393>
actor = 'terry.reedy'
assignee = 'terry.reedy'
closed = True
closed_date = <Date 2017-07-20.05:02:15.289>
closer = 'terry.reedy'
components = ['IDLE', 'Tests', 'Windows']
creation = <Date 2017-07-19.11:28:53.389>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 30968
keywords = ['buildbot']
message_count = 13.0
messages = ['298668', '298679', '298695', '298697', '298701', '298702', '298706', '298711', '298735', '298737', '298741', '298758', '298759']
nosy_count = 7.0
nosy_names = ['terry.reedy', 'paul.moore', 'vstinner', 'tim.golden', 'zach.ware', 'steve.dower', 'louielu']
pr_nums = ['2753', '2769', '5508']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue30968'
versions = ['Python 3.7']
```

</p></details>

**Comment by vstinner** [comment]
The build only contains one change:

commit f776eb0f0e046f2fa3a96540bb42d8cf970f6c55

bpo-30917: IDLE: Add config.IdleConf unittests (bpo-2691) Patch by Louie Lu.

http://buildbot.python.org/all/builders/x86%20Windows7%203.x/builds/893/steps/test/logs/stdio

\======================================================================
FAIL: test_get_font (idlelib.idle_test.test_config.IdleConfTest)
\----------------------------------------------------------------------
```pytb
Traceback (most recent call last):
  File "D:\cygwin\home\db3l\buildarea\3.x.bolen-windows7\build\lib\idlelib\idle_test\test_config.py", line 605, in test_get_font
    (f['family'], 10 if f['size'] < 10 else f['size'], f['weight']))
AssertionError: Tuples differ: ('Courier New', 9, 'normal') != ('Courier New', 10, 'normal')
```


First differing element 1:
9
10

- ('Courier New', 9, 'normal')
?                 ^

\+ ('Courier New', 10, 'normal')
?                 ^^

**Comment by 9da18a29-a07d-4828-a801-24cc911246de** [comment]
After checking source code, I found that is my fault on the test code. After the blocker [PR 2754](https://github.com/python/cpython/pull/2754) been merged, I'll fix this issue.

**Comment by terryjreedy** [comment]
I merged [pr2754](https://github.com/python/cpython/pull/2754) in [pr2753](https://github.com/python/cpython/pull/2753), the 3.6 backport of [pr2691](https://github.com/python/cpython/pull/2691).  To avoid breaking 3.6 buildbots more than one test, I will hold off merging [pr2753](https://github.com/python/cpython/pull/2753) until a fix is available.

**Comment by terryjreedy** [comment]
Louie, I am fixing this (I believe).

**Comment by terryjreedy** [comment]
New changeset 9f9192afbb4e45d09f0d3d717b457d157dc46398 by terryjreedy in branch 'master':
 	bpo-30968: Fix test_get_font in IDLE's test_config.  (bpo-2769)
https://github.com/python/cpython/commit/9f9192afbb4e45d09f0d3d717b457d157dc46398

**Comment by 9da18a29-a07d-4828-a801-24cc911246de** [comment]
Yes, Terry's patch fixed this.

**Comment by terryjreedy** [comment]
Fix worked: test_idle passed in build 897.
http://buildbot.python.org/all/builders/x86%20Windows7%203.x/builds/897/steps/test/logs/stdio
I also included the fix of [PR2769](https://github.com/python/cpython/pull/2769) in [PR2753](https://github.com/python/cpython/pull/2753) (3.6 backport).

**Comment by vstinner** [comment]
Terry: FYI your commit message starts with multiple spaces. I would also suggest you to format your name as you did previously: "Terry Jan Reedy".

commit 9f9192afbb4e45d09f0d3d717b457d157dc46398
Author: terryjreedy \<tjreedy@udel.edu\>
Date:   Wed Jul 19 20:45:36 2017 -0400

            bpo-30968: Fix test_get_font in IDLE's test_config.  (bpo-2769)

**Comment by terryjreedy** [comment]
I don't understand 'format your name' as there is no formatting that I did.  Github copies account name and email as they are.

I have no idea about commit message.  The PR title has no extra spaces.  But github does not always copy correctly.

**Comment by terryjreedy** [comment]
I will try to look at the first line more carefully before hitting the 2nd green button.

**Comment by vstinner** [comment]
Terry J. Reedy: I don't understand 'format your name' as there is no formatting that I did.

GitHub picks your commits unchanged. Git picks your name from your .gitconfig configuration file, global $HOME/.gitconfig or local \<cpython\>/.gitconfig.

It's just that I saw "terryjreedy" in your last commit, instead of the usual "Terry Jan Reedy".

It's up to you to pick whatever name you want, I was just surprised to see the name formatted differently.

> I have no idea about commit message.  The PR title has no extra spaces.  But github does not always copy correctly.

It doesn't matter at all. It's just that I was surprised ;-)

**Comment by 9da18a29-a07d-4828-a801-24cc911246de** [comment]
For the problem about the small name in git log Author, if you want to changed it, it will need to setup at https://github.com/settings/profile and update the first column "Name".

I think this is because we merged on GitHub, and GitHub ignores the git author but use GitHub account's name (if this didn't setup, it will fallback to GitHub account). I have some small name with "mlouielu" when I changed to this new account.
